### DISCUSSION TITLE
Boundary Conditions/Sourcing with Adiabatic Equation of State

### DISCUSSION THREAD

**Discussion by hwhitehead** [PrincetonUniversity/athena#527] [Q&A] [answered]
When an adiabatic equation of state is used (without magnetic fields), do the boundary conditions need to set ALL primitive variables (rho, v, P), or just rho and v (as in the isothermal case)?

Similarly, when constructing the initial domain, does the cell energy also need to be specified (instead of just rho, M as in isothermal)?

When custom source terms are included, am I correct in saying that any source terms which update momenta should NOT update the cell energies (cons(IEN) values), due to the barotropic equation of state?

Many thanks.

**Comment by c-white** [comment, marked as answer]
If using a non-barotropic (e.g. adiabatic) equation of state, the gas has 5 degrees of freedom, and so all 5 must be set whenever any are. This includes pressure in boundary conditions, total energy density in initial conditions, and changes to total energy density in source terms.

For the source term, if you update momentum, you affect the implied velocity. This in turn changes the implied kinetic energy density. If this is not taken into account by modifying the total energy density, the code will subtract the new kinetic energy density from the old total energy density and infer the wrong internal energy density. That is, ignoring energy leads to spurious heating or cooling that is probably not intended.

**Reply by hwhitehead** [reply]
Cheers Chris, to check my understanding:

- For a baroptropic equation of state, the cell energy should always be changed to match the change in kinetic energy, so as to maintain the internal energy (which is set only by the density)
- For a non-barotropic equation of state, it is my understanding that the cell energy should be changed slightly differently. From inspection of the source terms already in Athena++ (such as the shearing forces), an external acceleration $a_\text{ext}$ results in an energy change $\rho v \cdot a_\text{ext}$. Does this hold for general accelerations with non-barotropic equations of state?

**Reply by c-white** [reply]
For the barotropic case, energy and pressure are not tracked. In fact, they are not even allocated, so you should not try to set them.

Yes, the kinetic energy density changes by $\Delta k = \Delta t \rho v \cdot a$ in general.

**Reply by hwhitehead** [reply]
Thanks Chris. One last question, in your original response, you state that the adiabatic equation of state is non-barotropic. 
$P = K \rho^\gamma$
I'm slightly confused as to why this would be the case, is it because the adiabatic constant K needs to be specified? I would have assumed that once K is set, the EoS IS barotropic.

**Reply by c-white** [reply]
Athena uses "adiabatic" to mean the gas expands and contracts adiabatically with $p / \rho^\gamma$ constant *in the absence of shocks*. Indeed, shocks change $K$, so it is not constant in time, nor need it be constant in space in the initial conditions. You can think of specifying $\rho$ and $K$, but the code is written in the mindset of $\rho$ and $p$ being the two parameters that describe the thermodynamic state of the gas.

On the other hand, Athena uses "isothermal" to mean $p / \rho$ is unconditionally constant in space and time. That is, any effects of shocks or even adiabatic heating/cooling from compression/expansion, is assumed to be undone by a balance of optically thin radiative heating/cooling on timescales much shorter than the timestep.

**Reply by hwhitehead** [reply]
This makes much more sense now, I forgot about the effect of shocks. Thanks again!
